docs/DeveloperGuide.md

```diff
@@ -100,14 +100,30 @@ The boolean return value of `parseInput()` indicates whether the user has chosen
 
 ### Implementation
 
+#### Student
+
+The `Student` class has a composition relationship with class StudentList. Its purpose is to store key information on the different students that have applied for the Student Exchange Program. Such information include their GPA and university preferences, which helps us allocate them to the various universities fairly, and also other information which helps the app track their allocation status.
+
+#### StudentList
+
+The ```StudentList``` is a fundamental component which is initiated as soon as FindOurSEP is initialised. Its purpose is to hold the necessary information of the different students that are applying for SEP. By having the list of students we are able to fairly allocate universities to the different students by comparing them to their cohort.
+
+The sequence below illustrates the interactions between ```StudentList``` and ```Student``` when and addCommand is called with the appropriate inputs.
+
+![StudentListSequence](UML_Diagrams/StudentList.drawio.svg)
```

Comment:
Do you think it's better to show the interactions between the StudentList class and the Student class in the addStudent() method? Maybe through the use of a loop frame?

Reply:
I don't think there is a loop frame since it is only adding 1 student

Reply:
There's actually a for loop used to check if the Student already exist within StudentList. Feel free to disregard my comments if this is too trivial :)